components/aws/sagemaker/tests/integration_tests/utils/kfp_client_utils.py

```diff
@@ -54,6 +56,30 @@ def compile_run_monitor_pipeline(
     timeout,
     status_to_check="succeeded",
     check=True,
+):
+    try:
+        return pipeline_run(client, experiment_id, pipeline_definition, input_params, output_file_dir, pipeline_name, timeout,
+                            status_to_check="succeeded",
+                            check=True)
+    except ApiException as api_exception:
+        kfp_installed_namespace = utils.get_kfp_namespace()
+        client = kfp.Client(namespace=kfp_installed_namespace)
+        if api_exception.status < 500:
+            raise api_exception
+        return pipeline_run(client, experiment_id, pipeline_definition, input_params, output_file_dir, pipeline_name, timeout,
+                            status_to_check="succeeded",
+                            check=True)    
+
+def pipeline_run(
+    client,
+    experiment_id,
+    pipeline_definition,
+    input_params,
+    output_file_dir,
+    pipeline_name,
+    timeout,
+    status_to_check="succeeded",
```

Comment:
status_to_check will always be "succeeded" in this case.

https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/tests/integration_tests/resources/config/image-classification-groundtruth/config.yaml#L4